### ISSUE TITLE
Make extract-math conform to pandoc standards.

### ISSUE THREAD

**Issue by zkBounty** [vberlier/extract-math#82]
https://pandoc.org/MANUAL.html#math

![image](https://user-images.githubusercontent.com/52396134/85657493-f6196400-b666-11ea-8ab0-ee17825588e5.png)

Right now, the following fails. The string between the 2 dollar symbols gets output by the extract-math library as math, when it should.

John had $5 in his pocket while Jane had $6.

It's an extremely common usecase, that others like StackEdit seem to handle very nicely, and would be great to see extract-math also making it work.

**Comment by yashkothari** [comment]
I've been looking for a fix for this for a really long time too! @vberlier pls help 🙏🏼

**Comment by vberlier** [comment]
Okay I see I'm gonna try to add two options for this. They're going to be enabled by default I'm thinking something like this:

```js
{
  escape: '\\',
  delimiters: {
    inline: ['$', '$'],
    display: ['$$', '$$']
  },
  treatAsText: [/\$[0-9]/],
  allowSurroundingSpace: ['display']
}
```

**Comment by vberlier** [comment]
Alright so in the end I decided to only add the `allowSurroundingSpace` option, but I still added a check to make sure that the closing delimiter wasn't followed by a number before interpreting the segment as math. Now everything should work just like in pandoc. I released this in version 1.2.0.

**Comment by vberlier** [comment]
Actually there was a problem. I changed the way I'm generating the regex and now it works as expected. The patch is in 1.2.1.
